crates/store/src/state.rs

```diff
@@ -234,7 +234,9 @@ impl State {
 
         let header = block.header();
 
-        let tx_commitment = BlockHeader::compute_tx_commitment(block.transactions());
+        let tx_commitment = BlockHeader::compute_tx_commitment(
+            block.transactions().as_slice().iter().map(|tx| (tx.id(), tx.account_id())),
+        );
```

Comment:
Similar to the above comments. I would have probably added this logic to the `OrderedTransactions` so that here we could say:
```rust
let tx_commitment = block.transactions.commitment();
```